### ISSUE TITLE
Unable to insert middleware error when running v2 in prod.

### ISSUE THREAD

**Issue by tony-gutierrez** [houdiniproject/houdini#524]
## Description
Added env vars. 
ran `bin/rails server -e production`

## Expected Behavior
Server starts

## Actual Behavior
Got: 
```
	11: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/initializable.rb:32:in `run'
	10: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/initializable.rb:32:in `instance_exec'
	 9: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/application/finisher.rb:108:in `block in <module:Finisher>'
	 8: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/engine.rb:521:in `app'
	 7: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/engine.rb:521:in `synchronize'
	 6: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/engine.rb:524:in `block in app'
	 5: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/configuration.rb:94:in `merge_into'
	 4: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/configuration.rb:94:in `each'
	 3: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/configuration.rb:95:in `block in merge_into'
	 2: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3/lib/rails/configuration.rb:53:in `block in insert_before'
	 1: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3/lib/action_dispatch/middleware/stack.rb:101:in `insert'
/usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3/lib/action_dispatch/middleware/stack.rb:162:in `assert_index': No such middleware to insert before: ActionDispatch::Static (RuntimeError)
```

## Your Environment
* Ruby version: 2.7.2
* Server type and version: unbuntu 20.4
* Operating System and version: unbuntu 20.4
* Link to your project:

**Comment by wwahammy** [comment]
Hmmm this is odd. Did you set the `RAILS_SERVE_STATIC_FILES` environment variable?

**Comment by tony-gutierrez** [comment]
I didn't. I set the vars I found in env.template

**Comment by iRunner** [comment]
I get this: 

```
postgres@ubuntu-s-1vcpu-1gb-nyc3-01:~/houdini$ bin/rails server -e production
=> Booting Puma
=> Rails 6.1.3.1 application starting in production
=> Run `bin/rails server --help` for more startup options
Exiting
Traceback (most recent call last):
	88: from bin/rails:7:in `<main>'
	87: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:332:in `require'
	86: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:299:in `load_dependency'
	85: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:332:in `block in require'
	84: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	83: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	82: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	81: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	80: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	79: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/commands.rb:18:in `<main>'
	78: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/command.rb:50:in `invoke'
	77: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/command/base.rb:69:in `perform'
	76: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
	75: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
	74: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
	73: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/commands/server/server_command.rb:135:in `perform'
	72: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/commands/server/server_command.rb:135:in `tap'
	71: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/commands/server/server_command.rb:144:in `block in perform'
	70: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/commands/server/server_command.rb:39:in `start'
	69: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/server.rb:311:in `start'
	68: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/server.rb:379:in `handle_profiling'
	67: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/server.rb:312:in `block in start'
	66: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/server.rb:422:in `wrapped_app'
	65: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/server.rb:249:in `app'
	64: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/server.rb:349:in `build_app_and_options_from_config'
	63: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/builder.rb:66:in `parse_file'
	62: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/builder.rb:105:in `load_file'
	61: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/builder.rb:116:in `new_from_string'
	60: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/rack-2.2.3/lib/rack/builder.rb:116:in `eval'
	59: from config.ru:5:in `block in <main>'
	58: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:53:in `require_relative'
	57: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:332:in `require'
	56: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:299:in `load_dependency'
	55: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:332:in `block in require'
	54: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:34:in `require'
	53: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	52: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	51: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	50: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	49: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	48: from /var/lib/postgresql/houdini/config/environment.rb:12:in `<main>'
	47: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/application.rb:384:in `initialize!'
	46: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/initializable.rb:60:in `run_initializers'
	45: from /usr/local/rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/tsort.rb:205:in `tsort_each'
	44: from /usr/local/rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/tsort.rb:226:in `tsort_each'
	43: from /usr/local/rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/tsort.rb:347:in `each_strongly_connected_component'
	42: from /usr/local/rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/tsort.rb:347:in `call'
	41: from /usr/local/rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/tsort.rb:347:in `each'
	40: from /usr/local/rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/tsort.rb:349:in `block in each_strongly_connected_component'
	39: from /usr/local/rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/tsort.rb:431:in `each_strongly_connected_component_from'
	38: from /usr/local/rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	37: from /usr/local/rvm/rubies/ruby-2.7.2/lib/ruby/2.7.0/tsort.rb:228:in `block in tsort_each'
	36: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/initializable.rb:61:in `block in run_initializers'
	35: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/initializable.rb:32:in `run'
	34: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/initializable.rb:32:in `instance_exec'
	33: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/application/finisher.rb:129:in `block in <module:Finisher>'
	32: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:51:in `run_load_hooks'
	31: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:51:in `each'
	30: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:52:in `block in run_load_hooks'
	29: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:66:in `execute_hook'
	28: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:61:in `with_execution_control'
	27: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/lazy_load_hooks.rb:68:in `block in execute_hook'
	26: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/devise-4.7.3/lib/devise/rails.rb:17:in `block in <class:Engine>'
	25: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/application.rb:169:in `reload_routes!'
	24: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/application/routes_reloader.rb:21:in `reload!'
	23: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/application/routes_reloader.rb:47:in `load_paths'
	22: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/application/routes_reloader.rb:47:in `each'
	21: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/railties-6.1.3.1/lib/rails/application/routes_reloader.rb:47:in `block in load_paths'
	20: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:326:in `load'
	19: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:299:in `load_dependency'
	18: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activesupport-6.1.3.1/lib/active_support/dependencies.rb:326:in `block in load'
	17: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:59:in `load'
	16: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:59:in `load'
	15: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activestorage-6.1.3.1/config/routes.rb:3:in `<main>'
	14: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/route_set.rb:409:in `draw'
	13: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/route_set.rb:427:in `eval_block'
	12: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/route_set.rb:427:in `instance_exec'
	11: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activestorage-6.1.3.1/config/routes.rb:4:in `block in <main>'
	10: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/mapper.rb:901:in `scope'
	 9: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/activestorage-6.1.3.1/config/routes.rb:5:in `block (2 levels) in <main>'
	 8: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/mapper.rb:718:in `get'
	 7: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/mapper.rb:765:in `map_method'
	 6: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/mapper.rb:1663:in `match'
	 5: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/mapper.rb:1915:in `map_match'
	 4: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/mapper.rb:1915:in `each'
	 3: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/mapper.rb:1921:in `block in map_match'
	 2: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/mapper.rb:1957:in `decomposed_match'
	 1: from /usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/mapper.rb:1986:in `add_route'
/usr/local/rvm/gems/ruby-2.7.2@fundraising/gems/actionpack-6.1.3.1/lib/action_dispatch/routing/route_set.rb:587:in `add_route': Invalid route name, already in use: 'rails_service_blob'  (ArgumentError)
You may have defined two routes with the same name using the `:as` option, or you may be overriding a route already defined by a resource with the same naming. For the latter, you can restrict the routes created with `resources` as explained here:
https://guides.rubyonrails.org/routing.html#restricting-the-routes-created
```

Ubuntu 20.04

**Comment by BenSturmfels** [comment]
Setting `RAILS_SERVE_STATIC_FILES` does resolve the `No such middleware` error for me. Does this indicate that there's a package missing that should be in the `:production` group in Gemfile? (I'm not familiar with Rails)

I'm also seeing the second issue `You may have defined two routes`. I'm essentially running the `bin/setup` as per `README.md`, sourcing `.env`, setting `DEVISE_SECRET_KEY` and running `bin/rails server -e production`.

**Comment by wwahammy** [comment]
> Setting `RAILS_SERVE_STATIC_FILES` does resolve the `No such middleware` error for me. Does this indicate that there's a package missing that should be in the `:production` group in Gemfile? (I'm not familiar with Rails)
> 
> I'm also seeing the second issue `You may have defined two routes`. I'm essentially running the `bin/setup` as per `README.md`, sourcing `.env`, setting `DEVISE_SECRET_KEY` and running `bin/rails server -e production`.

Hi @BenSturmfels and @iRunner: if you are running Rails in production and serving the files from your server instead of some other service (like Amazon S3 or some CDN), you need to set the RAILS_SERVE_STATIC_FILES environment variable to something. The Rails folks generally don't recommend that for performance reasons but that's your option.

The "You have defined two routes" error is something I haven't seen before. Please file an issue with as much detail as you have.

**Comment by wwahammy** [comment]
Also, please add something to the README about RAILS_SERVE_STATIC_FILES. Thanks all!

Also, don't be strangers and introduce yourselfs in the discussion forum! https://github.com/houdiniproject/houdini/discussions/675

**Comment by BenSturmfels** [comment]
> > Setting `RAILS_SERVE_STATIC_FILES` does resolve the `No such middleware` error for me. Does this indicate that there's a package missing that should be in the `:production` group in Gemfile? (I'm not familiar with Rails)
> > I'm also seeing the second issue `You may have defined two routes`. I'm essentially running the `bin/setup` as per `README.md`, sourcing `.env`, setting `DEVISE_SECRET_KEY` and running `bin/rails server -e production`.
> 
> Hi @BenSturmfels and @iRunner: if you are running Rails in production and serving the files from your server instead of some other service (like Amazon S3 or some CDN), you need to set the RAILS_SERVE_STATIC_FILES environment variable to something. The Rails folks generally don't recommend that for performance reasons but that's your option.

Silly question perhaps as a Rails noob, but if `RAILS_SERVE_STATIC_FILES` isn't set, shouldn't it just 404 when asked for a static file? There shouldn't be an unhandled error or server startup right? Assuming I did want to offload the static files to Nginx, how would I start it up so it doesn't error? I'm don't have any issues with offloading static files - just looking to spin up a server with minimal effort to try Houdini out.

**Comment by wwahammy** [comment]
No, you're right. There must be some middleware causing some issue. I'm not
sure honestly.  Searching the rails source code for that environment
variable might help but I'm really not sure off hand.

**Comment by wwahammy** [comment]
Hey all, the immediate workaround is to comment out this line in the Gemfile:

```ruby
gem 'font_assets', '~> 0.1.14' # for serving fonts on cdn https://github.com/ericallam/font_assets
```

Font assets was used in the past for an older version of Rails. I'm not sure offhand what breaks if you keep font_assets off. Please report differences between development and production as it relates to fonts.

**Comment by BenSturmfels** [comment]
@wwahammy thanks very much for the quick response!

**Comment by BenSturmfels** [comment]
Closing as this has resolved in #1054.
